src/scenes/Search/store.js

```diff
@@ -30,15 +30,14 @@ export default class SearchStore {
   init = async () => {
     this.topicAddrs = new Map();
     this.loading = true;
-    this.list = await this.fetch(this.phrase);
+    await this.fetch(this.phrase);
     runInAction(() => {
       this.loading = false;
       /** the following is a temp work around as searchAll in backend, since we don't have a combined type of Oracle and Topic */
-      this.withdraws = this.list.filter(event => event.phase === Phases.WITHDRAWING && this.topicAddrs.set(event.address, true));
-      this.finalizes = this.list.filter(event => event.phase === Phases.FINALIZING && !this.topicAddrs.has(event.topicAddress) && this.topicAddrs.set(event.topicAddress, true));
-      this.sets = this.list.filter(event => event.phase === Phases.RESULT_SETTING && !this.topicAddrs.has(event.topicAddress) && this.topicAddrs.set(event.topicAddress, true));
-      this.votes = this.list.filter(event => event.phase === Phases.VOTING && event.status === 'VOTING' && !this.topicAddrs.has(event.topicAddress) && this.topicAddrs.set(event.topicAddress, true));
-      this.bets = this.list.filter(event => event.phase === Phases.BETTING && event.status === 'VOTING' && !this.topicAddrs.has(event.topicAddress));
+      this.finalizes = this.oracles.filter(event => event.phase === Phases.FINALIZING && !this.topicAddrs.has(event.topicAddress) && this.topicAddrs.set(event.topicAddress, true));
```

Comment:
Another way you could find the latest oracle is to sort it by `blockNum DESC` and get the first item. The higher the `blockNum` the more recent it is..

Reply:
I have changed the logic, plz refer to line 68 and 71, and then 33-36.
Basically I'm using `orderBy` of `blockNum DESC` instead of `endTime`, then I use `lodash uniqBy` of `topicAddress` to remove out-dated duplicates. After that, I sort it by `endTime`. Then I filter out each phase only according to the `event.phase` and `event.status`, getting rid of `HashMap`.

This way is much cleaner, thx for the tip!

FYI, I noticed that our backend supports `orderBy` in `Array`, but our frontend can only handle a single `orderBy` object because of the way how `formatObject` in `graphql/queries.js` parses objects. Do you think it deserves an issue? Although currently a single `orderBy` is enough.

FYI2, our search result page currently is not dynamic as the page won't change if an event transits into another phase. Do you think we should make it dynamic?

Reply:
` syncBlockNum` works like a charm!

I now realize how handy block number is, thx!